app/controllers/web_hooks/slack_controller.rb

```diff
@@ -23,9 +23,17 @@ def build_create_response
     when 'url_verification'
       params[:challenge]
     when 'event_callback'
-      "Add Slack message event to db #{slack_message(params)}" if params[:event][:type] == 'message'
+      check_event_subtype(params)
     else
       "Error"
     end
   end
+
+  def check_event_subtype(params)
+    subtypes = %w[file_comment file_mention file_share me_message message_replied thread_broadcast]
+    if params[:event][:type] == 'message' &&
+       (params[:event][:subtype].nil? || subtypes.include?(params[:event][:subtype]))
+      "Add Slack message event to db #{slack_message(params)}"
```

Comment:
Ah ok. Missed this originally. I think its a little odd to trigger a database insertion from within string interpolation. What about something like:
```
when 'event_callback'
   event = params[:event]
   return "Skipping disallowed event #{event[:type]:#{event[:subtype]" unless allow_event?(event)
   result = create_slack_message(event)
   "Add Slack message event to db #{result}"
```